ktor-http/common/src/io/ktor/http/Query.kt

```diff
@@ -40,27 +40,42 @@ private fun ParametersBuilder.parse(query: String, startIndex: Int, limit: Int)
     if (count == limit) {
         return
     }
-    appendParam(query, nameIndex, equalIndex, query.length)
+    appendParam(query, nameIndex, equalIndex, query.length, decode)
 }
 
-private fun ParametersBuilder.appendParam(query: String, nameIndex: Int, equalIndex: Int, endIndex: Int) {
+private fun ParametersBuilder.appendParam(
+    query: String,
+    nameIndex: Int,
+    equalIndex: Int,
+    endIndex: Int,
+    decode: Boolean
+) {
     if (equalIndex == -1) {
         val spaceNameIndex = trimStart(nameIndex, endIndex, query)
         val spaceEndIndex = trimEnd(spaceNameIndex, endIndex, query)
 
         if (spaceEndIndex > spaceNameIndex) {
-            val name = query.decodeURLQueryComponent(spaceNameIndex, spaceEndIndex)
+            val name = when {
+                decode -> query.decodeURLQueryComponent(spaceNameIndex, spaceEndIndex)
+                else -> query.substring(spaceNameIndex, spaceEndIndex)
+            }
             appendAll(name, emptyList())
         }
     } else {
```

Comment:
I would put early return instead of else

Reply:
This is the unchanged code. I don't want to do a clean up of the old code here to not bloat PR, since it's very big already